docs: correct IPAM mode name in BGP control plane installation docs

tklauser opened the pull request:
The IPAM mode is currently called "cluster-pool-v2beta", not
"cluster-pool-v2".

**_Note for backporters_**: the file was renamed from `Documentation/gettingstarted/bgp-control-plane.rst` in 69d07f79cb17dd0a543043152a32604bb4226ee3. The `v1.12` branch still has the old name.

tklauser committed:
docs: correct IPAM mode name in BGP control plane installation docs

The IPAM mode is currently called "cluster-pool-v2beta", not
"cluster-pool-v2".

Signed-off-by: Tobias Klauser <tobias@cilium.io>

qmonnet commented on Documentation/installation/bgp-control-plane.rst:
```diff
@@ -297,7 +297,7 @@ error is logged and the ``BGPRouterManager`` will continue to the next
 
 GoBGP BGPRouterManager Architecture
-                                   
+
```
This looks like there should be a title underline here?

The formatting on this whole section is messed up. All sentences are separated by empty lines. Would you mind reformatting the paragraphs and item list (above) correctly? I can do as a follow-up otherwise.

tklauser replied:
Sure, added in a second commit.

ldelossa reviewed (approved):
Thanks for the fixup.

qmonnet commented on Documentation/installation/bgp-control-plane.rst:
```diff
@@ -275,55 +275,37 @@ GoBGP Implementation
+package. You can find this implementation in ``pkg/bgpv1/gobgp``.
 
-You can find this implementation in ``pkg/bgpv1/gobgp``.
+This implementation will.
```
```suggestion
This implementation will:
```

qmonnet reviewed (approved):
Thanks a lot!

tklauser committed:
docs: fix formatting in BGP control plane installation docs

Add missing title underline and format paragraphs/lists correctly.

Suggested-by: Quentin Monnet <quentin@isovalent.com>
Signed-off-by: Tobias Klauser <tobias@cilium.io>
